tests/0098-consumer-txn.cpp

```diff
@@ -0,0 +1,851 @@
+                         "Expected 13, got: "
+                      << msgs.size());
+  delete_messages(msgs);
+  
+  Test::delete_topic(c, topic_name.c_str());
+
+  c->close();
+  delete c;
+
+
+  // Test 4.3
+
+  topic_name = Test::mk_topic_name("0098-consumer_txn-4.3", 1);
+  c = create_consumer(topic_name, "READ_COMMITTED");
+  Test::create_topic(c, topic_name.c_str(), 1, 3);
+
+  execute_java_produce_cli(bootstrap_servers, topic_name, "4.3");
+
+  msgs = consume_messages(c, topic_name, 0);
+  test_assert(msgs.size() == 1,
+              tostr() << "Consumed unexpected number of messages. "
+                         "Expected 7, got: "
+                      << msgs.size());
+  delete_messages(msgs);
+  c->close();
+  delete c;
+
+  c = create_consumer(topic_name, "READ_UNCOMMITTED");
+  msgs = consume_messages(c, topic_name, 0);
+  test_assert(msgs.size() == 13,
+              tostr() << "Consumed unexpected number of messages. "
+                         "Expected 13, got: "
+                      << msgs.size());
+  delete_messages(msgs);
+
+  Test::delete_topic(c, topic_name.c_str());
+
+  c->close();
+  delete c;
+
+
+  // Test 5 - split transaction across message set.
+
+  topic_name = Test::mk_topic_name("0098-consumer_txn-5", 1);
+  c = create_consumer(topic_name, "READ_COMMITTED");
+  Test::create_topic(c, topic_name.c_str(), 1, 3);
+
+  execute_java_produce_cli(bootstrap_servers, topic_name, "5");
+
+  msgs = consume_messages(c, topic_name, 0);
+  test_assert(msgs.size() == 9,
+              tostr() << "Consumed unexpected number of messages. "
+                         "Expected 9, got: "
+                      << msgs.size());
+  delete_messages(msgs);
+
+  Test::delete_topic(c, topic_name.c_str());
+
+  c->close();
+  delete c;
+
+
+  // Test 6 - transaction left open
+
+  topic_name = Test::mk_topic_name("0098-consumer_txn-0", 1);
+  c = create_consumer(topic_name, "READ_COMMITTED");
+  Test::create_topic(c, topic_name.c_str(), 1, 3);
+
+  execute_java_produce_cli(bootstrap_servers, topic_name, "6");
+
+  msgs = consume_messages(c, topic_name, 0);
+  test_assert(msgs.size() == 1,
+              tostr() << "Consumed unexpected number of messages. "
+                         "Expected 1, got: "
+                      << msgs.size());
+
+  test_assert(partition_0_ls_offset + 3 == partition_0_hi_offset,
+              tostr() << "Expected hi_offset to be 3 greater than ls_offset "
+                         "but got hi_offset: "
+                      << partition_0_hi_offset
+                      << ", ls_offset: "
+                      << partition_0_ls_offset);
+
+  delete_messages(msgs);
+
+  Test::delete_topic(c, topic_name.c_str());
+
+  c->close();
+  delete c;
+}
+
+extern "C" {
+  int main_0098_consumer_txn (int argc, char **argv) {
+    if (test_quick) {
+      Test::Skip("Test skipped due to quick mode\n");
+      return 0;
+    }
+    
+    do_test_consumer_txn_test();
```

Comment:
Instead of skipping this entire test suite, it is better if you cut down on the number of tests but still run the most important ones.

Reply:
yes, but that means i need to work out how to get the java application built prior to the tests being run, and I was avoiding that effort.

Reply:
Isn't that just a `make -C tests/java`?

Reply:
A kafka distribution is needed for the java client dependency - that doesn't seemed to be something that's installed already, so I'm still skipping these tests in test_quick mode still